src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -31,12 +32,26 @@ public class CryptoConfig
 
         private static volatile Dictionary<string, string> s_defaultOidHT = null;
         private static volatile Dictionary<string, object> s_defaultNameHT = null;
+        private static volatile Dictionary<string, Type> appNameHT = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+        private static volatile Dictionary<string, string> appOidHT = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
         private static readonly char[] SepArray = { '.' }; // valid ASN.1 separators
 
         // CoreFx does not support AllowOnlyFipsAlgorithms
         public static bool AllowOnlyFipsAlgorithms => false;
 
+        // Private object for locking instead of locking on a public type for SQL reliability work.
+        private static Object s_InternalSyncObject;
+        private static Object InternalSyncObject {
+            get {
+                if (s_InternalSyncObject == null) {
```

Comment:
The lazy initialization is probably unnecessary on Core since CryptoConfig isn't on the same hot paths that it was on NetFX. I'd suggest just initializing the field.